AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -557,122 +365,6 @@ final public class AWSMobileClient: _AWSMobileClient {
-        
-        self.getAWSCredentials { (credentials, error) in
-            if let credentials = credentials {
-                credentialsTaskCompletionSource.set(result: credentials)
-            } else if let error = error {
-                credentialsTaskCompletionSource.set(error: error)
-            }
-        }
-        return credentialsTaskCompletionSource.task
-    }
-    
-    
-    /// Invalidates the cached temporary AWS credentials. If the credentials provider does not cache temporary credentials, this operation is a no-op.
-    override public func invalidateCachedTemporaryCredentials() {
-        self.internalCredentialsProvider?.invalidateCachedTemporaryCredentials()
-    }
-    
-    
-    /// Get/retrieve the identity id for this provider. If an identity id is already set on this provider, no remote call is made and the identity will be returned as a result of the AWSTask (the identityId is also available as a property). If no identityId is set on this provider, one will be retrieved from the service.
-    ///
-    /// - Returns: Asynchronous task which contains the identity id or error.
-    override public func getIdentityId() -> AWSTask<NSString> {
-        guard self.internalCredentialsProvider != nil else {
-            return AWSTask(error: AWSMobileClientError.cognitoIdentityPoolNotConfigured(message: "Cannot get identityId since cognito credentials configuration is not available."))
-        }
-        let identityFetchTaskCompletionSource: AWSTaskCompletionSource<NSString> = AWSTaskCompletionSource()
-        self.internalCredentialsProvider?.getIdentityId().continueWith(block: { (task) -> Any? in
-            if let error = task.error {
-                if error._domain == AWSCognitoCredentialsProviderHelperErrorDomain
-                    && error._code == AWSCognitoCredentialsProviderHelperErrorType.identityIsNil.rawValue {
-                    identityFetchTaskCompletionSource.set(error: AWSMobileClientError.identityIdUnavailable(message: "Fetching identity id on another thread failed. Please retry by calling `getIdentityId()` method."))
-                } else {
-                    identityFetchTaskCompletionSource.set(error: error)
-                }
-            } else if let result = task.result {
-                identityFetchTaskCompletionSource.set(result: result)
-            }
-            return nil
-        })
-        
-        return identityFetchTaskCompletionSource.task
-    }
-    
-    
-    /// Clear the cached AWS credentials for this provider.
-    override public func clearCredentials() {
-        self.internalCredentialsProvider?.clearCredentials()
-    }
-    
-    
-    /// Clear ALL saved values for this provider (identityId, credentials, logins).
-    override public func clearKeychain() {
-        self.internalCredentialsProvider?.clearKeychain()
-    }
-    
 }
 
 // MARK:- AWSMobileClient Cognito configuration
```

Comment:
Would it make sense to move the below extension to AWSMobileClientExtensions?

Reply:
Will cleanup in later PR if required. Since it is part of configuration, I thought of keeping it in this main file.